google/cloud/bigquery/v2/minimal/mocks/mock_job_connection.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_BIGQUERY_V2_MINIMAL_MOCKS_MOCK_JOB_CONNECTION_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_BIGQUERY_V2_MINIMAL_MOCKS_MOCK_JOB_CONNECTION_H
+
+#include "google/cloud/bigquery/v2/minimal/internal/job_connection.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace bigquery_v2_minimal_mocks {
```

Comment:
The mock is a public API that uses `internal` symbols. It's not a big deal, but ideally we would keep it private until it is ready. (e.g. it might show up in our docs and confuse customers)

In my opinion, you should define the mocks in `bigquery_v2_minimal_internal` for now, with a `TODO(#XXXX)` to move it to `bigquery_v2_minimal_mocks` when the library is ready to go public.